circe-v1/src/main/scala/dsl/circe/CirceVaporsEncoders.scala

```diff
@@ -13,23 +13,35 @@ import sourcecode.{File, Line}
 
 trait CirceVaporsEncoders extends MidPrioritySourceInfoEncoders {
 
+  implicit val encodeExprStateEmpty: Encoder.AsObject[ExprState[Nothing, Nothing]] =
+    Encoder.AsObject.instance(_ => JsonObject.empty)
+
   implicit val encodeHasSourceCodeInfo: Encoder.AsObject[HasSourceCodeInfo] = Encoder.AsObject.instance { ctx =>
     val SourceCodeInfo(File(file), Line(line)) = ctx.debugSource
     JsonObject(
       "source" -> s"$file:$line".asJson,
     )
   }
 
-  implicit def encodeExprState[I : Encoder, O : Encoder]: Encoder.AsObject[ExprState[I, O]] =
-    Encoder.AsObject.instance { state =>
-      JsonObject(
-        "input" -> state.input.asJson,
-        "output" -> state.output.asJson,
-      )
-    }
+  // TODO: Define this properly
+  implicit val encodeEvidence: Encoder[Evidence] = Encoder.instance { evidence =>
+//    evidence.factSet.map(_.asJson)
+    ???
+  }
```

Comment:
I don't have time to figure this one out for the first milestone, but I want to solve this before v1.0.0. I think we need to pack the encoder into the `Justified` type (maybe adding an `OP` there?) Not sure, but when I tried this, the changes rippled throughout the code in a way that I wasn't comfortable with tackling just yet.

Reply:
I can just serialize this as a string for now.